frontend/components/Documents.js

```diff
@@ -56,6 +57,25 @@ const Documents = () => {
 	    }));
     };
 
+    const handleAttributeInputChange = (event, index) => {
+        const {name, value} = event.target;
+        setNewAttributes(prevAttributes => {
+            return prevAttributes.map((attribute, i) => {
+                return i === index ? {...attribute, [name]:value} : attribute;
```

Comment:
Is `{...attribute, [name]:value}` the correct data structure and transformation here? I see above when we create the `newAttributes` state, it's an array of objects of the shape `{"name": "", "value": ""}`. Here in line 64, it seems like we'd spread the default 'empty' `key: value` pairs and then add a new `key: value` pair (i.e., we'd end up with something like `{"name": "", "value": "", "newName": "newValue"}`). 

That is: when we update a default object in `newAttributes` based on user input, its name is set to the value of the `name` key and its value set to the value of the `value` key, but _not_ that its name and value are set to an actual `key: value` pairing in the object. Do I read that all correctly, and/or am I making any sense?

Reply:
I think the way I named the `name` variable might have called some confusion. The `name` variable actually refers to the name of the input tag, so it's possible values are `"name"` and `"value"`. Because we use the same function to handle the input changes for both `"name"` and `"value"`, the `{...attribute` in `{...attribute, [name]:value}` refers to the value we're not changing. For example, if we currently have `{"name": "color", "value": "blue"}` and we wanted to change the color to green, the variables will be `name = "value"` and `value = "green"` so then `...attribute` refers to `"name": "color"` and `[name]:value` refers to `"value":"green"`